schematics/table/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.ts

```diff
@@ -0,0 +1,132 @@
+    {position: 20, name: 'Calcium', weight: 40.078, symbol: 'Ca'},
+  ];
+
+  constructor(private _paginator: MatPaginator, private _sort: MatSort) {
+    super();
+  }
+
+  connect(): Observable<Element[]> {
+    const stream = [
+      of(this.data),
+      this._paginator.page,
+      this._sort.sortChange
+    ];
+
+    return merge(...stream).pipe(map(() => {
+      return this.getPagedData(this.getSortedData([...this.data]));
+    }));
+  }
+
+  disconnect() {}
+
+  getPagedData(data) {
+    const startIndex = this._paginator.pageIndex * this._paginator.pageSize;
+    return data.splice(startIndex, this._paginator.pageSize);
+  }
+
+  getSortedData(data) {
+    if (!this._sort.active || this._sort.direction === '') { return data; }
+    return data.sort((a, b) => {
+      const propertyA = a[this._sort.active];
+      const propertyB = b[this._sort.active];
+      const valueA = isNaN(+propertyA) ? propertyA : +propertyA;
+      const valueB = isNaN(+propertyB) ? propertyB : +propertyB;
+      return (valueA < valueB ? -1 : 1) * (this._sort.direction === 'asc' ? 1 : -1);
+    });
+  }
+}
+
+@Component({
+  selector: '<%= selector %>',<% if(inlineTemplate) { %>
+  template: `
+    <div class="mat-elevation-z8">
+      <mat-table #table [dataSource]="dataSource" matSort>
+
+        <!-- Position Column -->
+        <ng-container matColumnDef="position">
+          <mat-header-cell *matHeaderCellDef mat-sort-header>No.</mat-header-cell>
+          <mat-cell *matCellDef="let element">{{element.position}}</mat-cell>
+        </ng-container>
+
+        <!-- Name Column -->
+        <ng-container matColumnDef="name">
+          <mat-header-cell *matHeaderCellDef mat-sort-header>Name</mat-header-cell>
+          <mat-cell *matCellDef="let element">{{element.name}}</mat-cell>
+        </ng-container>
+
+        <!-- Weight Column -->
+        <ng-container matColumnDef="weight">
+          <mat-header-cell *matHeaderCellDef mat-sort-header>Weight</mat-header-cell>
+          <mat-cell *matCellDef="let element">{{element.weight}}</mat-cell>
+        </ng-container>
+
+        <!-- Color Column -->
+        <ng-container matColumnDef="symbol">
+          <mat-header-cell *matHeaderCellDef mat-sort-header>Symbol</mat-header-cell>
+          <mat-cell *matCellDef="let element">{{element.symbol}}</mat-cell>
+        </ng-container>
+
+        <mat-header-row *matHeaderRowDef="displayedColumns"></mat-header-row>
+        <mat-row *matRowDef="let row; columns: displayedColumns;"></mat-row>
+      </mat-table>
+
+      <mat-paginator #paginator
+        [length]="20"
+        [pageIndex]="0"
+        [pageSize]="10"
+        [pageSizeOptions]="[5, 10, 25, 100]">
+      </mat-paginator>
+    </div>
+  `,<% } else { %>
+  templateUrl: './<%= dasherize(name) %>.component.html',<% } if(inlineStyle) { %>
+  styles: []<% } else { %>
+  styleUrls: ['./<%= dasherize(name) %>.component.<%= styleext %>']<% } %><% if(!!viewEncapsulation) { %>,
```

Comment:
Is the `!!` necessary?

Reply:
I didn't write this, I copied the code from the default component template for this section.

Reply:
Doesn't mean it shouldn't be corrected.